### PR TITLE
[FR-115] NextJS Expandable Images

### PR DESCRIPTION
**Pull request by tylergeorges**

### CODE REVIEW

**Review by Yomanz — CHANGES_REQUESTED** [review]
The animation is way too slow and it's responsiveness isn't fantastic but not sure what you can do about that, the animation needs to be sped up though

File: src/components/Overlays/Modal/ImageModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        type="modal"
+        isOpen={isOpen}
+        onClick={closeImage}
+        onTransitionEnd={handleTransitionEnd}
+      />
+
+      <Styles.ModalContainerWrapper isOpen={isOpen} role="dialog">
+        <Styles.ModalContainer isOpen={isOpen} background="transparent">
+          <Styles.ImagePopup
+            width={width as number}
```
Comment (Yomanz) [inline src/components/Overlays/Modal/ImageModal/index.tsx:51]: Why do you have to cast them, what are they actually passed as?

File: src/components/Overlays/Modal/ImageModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            alt="Image Popup"
+            height={height as number}
```
Comment (Yomanz) [inline src/components/Overlays/Modal/ImageModal/index.tsx:53]: See above

File: src/components/Providers/MessageRenderer.tsx [hunk 3]
```diff
@@ -145,6 +149,10 @@ export const MessageRenderer = ({ children }: MessageRendererWrapperProps) => {
     }
   };
 
+  const handleImageClick = (image: APIAttachment | APIEmbed) => {
+    setImageModalData(image as APIEmbed['image']);
```
Comment (Yomanz) [inline src/components/Providers/MessageRenderer.tsx:153]: Is this a safe cast?

**Review by Yomanz — CHANGES_REQUESTED** [review]

File: src/components/Overlays/Modal/ImageModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [imageModalData]);
+
+  const closeImage = () => {
+    setToggled(false);
+  };
+
+  const isOpen = !!toggled;
```
Comment (advaith1) [inline src/components/Overlays/Modal/ImageModal/index.tsx:27]: isn't toggled always a boolean? why is this other variable needed, is `isOpen` not always equal to `toggled`?

File: src/components/Overlays/Modal/ImageModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+  if (!imageModalData) return <></>;
```
Comment (advaith1) [inline src/components/Overlays/Modal/ImageModal/index.tsx:29]:
```suggestion
  if (!imageModalData) return null;
```

**Review by advaith1 — CHANGES_REQUESTED** [review]
also the background dim needs to go away faster when the image is closed

**Review by Yomanz — APPROVED** [review]

**Comment by mason-rogers** [comment]
conflicts here @tylergeorges

**Review by mason-rogers — APPROVED** [review]

**Pull request merged**